chronos/transports/datagram.nim

```diff
@@ -28,7 +30,9 @@ type
     writer: Future[void]        # Writer vector completion Future
 
   DatagramCallback* = proc(transp: DatagramTransport,
-                           remote: TransportAddress): Future[void] {.gcsafe.}
+                           message: seq[byte],
```

Comment:
`openArray[byte]` tends to give more flexibility for the implementation.. a common option later on is to allow the original caller to set up a byte buffer of their choice..

Reply:
this is also the kind of place where `Result` is appropriate - giving either an error or a buffer but never both - makes for cleaner code everywhere

Reply:
@arnetheduck we can't use `openarray[T]` because it can't be captured inside of iterator, so it can't work in async procedures.